toolchain/check/import.cpp

```diff
@@ -113,56 +154,38 @@ auto AddImportNamespace(Context& context, SemIR::TypeId namespace_type_id,
           // id to get more accurate location.
           DiagnoseDuplicateName(context, name_id, import_id, prev_inst_id);
         }
-        return {.name_scope_id = namespace_inst->name_scope_id,
-                .inst_id = prev_inst_id,
+        return {.add_result = {.name_scope_id = namespace_inst->name_scope_id,
+                               .inst_id = prev_inst_id},
                 .is_duplicate_of_namespace_in_current_package = true};
       }
     }
   }
 
   auto import_id = make_import_id();
   CARBON_CHECK(import_id.has_value());
-  auto import_loc_id = context.insts().GetLocId(import_id);
 
-  auto namespace_inst =
-      SemIR::Namespace{.type_id = namespace_type_id,
-                       .name_scope_id = SemIR::NameScopeId::None,
-                       .import_id = import_id};
-  auto namespace_inst_and_loc =
-      import_loc_id.is_import_ir_inst_id()
-          ? MakeImportedLocIdAndInst(context, import_loc_id.import_ir_inst_id(),
-                                     namespace_inst)
-          // TODO: Check that this actually is an `AnyNamespaceId`.
-          : SemIR::LocIdAndInst(
-                Parse::AnyNamespaceId::UnsafeMake(import_loc_id.node_id()),
-                namespace_inst);
-  auto namespace_id =
-      AddPlaceholderInstInNoBlock(context, namespace_inst_and_loc);
-  context.import_ref_ids().push_back(namespace_id);
-  namespace_inst.name_scope_id =
-      context.name_scopes().Add(namespace_id, name_id, parent_scope_id);
-  ReplaceInstBeforeConstantUse(context, namespace_id, namespace_inst);
+  auto result = AddImportNamespaceAndNameResult{
```

Comment:
```suggestion
  AddImportNamespaceAndNameResult result = {
```

[Style](https://github.com/carbon-language/carbon-lang/blob/trunk/docs/project/cpp_style_guide.md#syntax-and-formatting): See the initialization section.

Reply:
Done.